src/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs

```diff
@@ -176,43 +176,32 @@ internal static Assembly LoadGeneratedAssembly(Type type, string defaultNamespac
                     {
                         serializerPath = Path.Combine(Path.GetDirectoryName(type.Assembly.Location), serializerName + ".dll");
                     }
-                }
-                catch
-                {
-                }
 
-                try
-                {
                     if ((string.IsNullOrEmpty(serializerPath) || !File.Exists(serializerPath)) && !string.IsNullOrEmpty(Assembly.GetEntryAssembly().Location))
                     {
                         serializerPath = Path.Combine(Path.GetDirectoryName(Assembly.GetEntryAssembly().Location), serializerName + ".dll");
                     }
+
+                    if (!string.IsNullOrEmpty(serializerPath))
+                    {
+                        serializer = Assembly.LoadFile(serializerPath);
+                    }
                 }
-                catch
-                {
-                }
- 
-                if (!string.IsNullOrEmpty(serializerPath))
+                catch (Exception e)
                 {
-                    try
+                    if (e is OutOfMemoryException)
                     {
-                        serializer = Assembly.LoadFile(serializerPath);
+                        throw;
                     }
-                    catch (Exception e)
+                    byte[] token = name.GetPublicKeyToken();
+                    if (token != null && token.Length > 0)
```

Comment:
This can be a separate issue. As we left out [Assembly.LoadWithPartialName(...)](http://referencesource.microsoft.com/#System.Xml/System/Xml/Serialization/Compilation.cs,188), looks like this block is doing nothing. Do you know why?